### PR TITLE
Add mqtt event platform

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Add mqtt event platform

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/96876
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/event.mqtt.markdown [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+```
+
+The event information is extracted from a JSON formatted MQTT message.
+To test, you can use the command line tool `mosquitto_pub` shipped with `mosquitto` or the `mosquitto-clients` package to send MQTT messages.
+
+To set trigger the `mqtt` event entity manually:
+
+```bash
+mosquitto_pub -h 127.0.0.1 -t home/doorbell/available -m "online"
+mosquitto_pub -h 127.0.0.1 -t home/doorbell/available -m '{"event_type": "hold"}'
```
Comment (frenck) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
mosquitto_pub -h 127.0.0.1 -t home/doorbell/state -m '{"event_type": "hold"}'
```

File: source/_integrations/event.mqtt.markdown [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+```
+
+Besides the required `event_type` element, the payload can contain additional event attributes.
+These additional attribute updates will be exposed as attributes on the `mqtt` event entity.
+
+The example below demonstrates how event attributes can be added to the event data.
+
+```bash
+mosquitto_pub -h 127.0.0.1 -t home/doorbell/available -m '{"event_type": "press", "duration": 0.1}'
```
Comment (frenck) [inline source/_integrations/event.mqtt.markdown]:
```suggestion
mosquitto_pub -h 127.0.0.1 -t home/doorbell/state -m '{"event_type": "press", "duration": 0.1}'
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/event.mqtt.markdown (hunk 1 above)
Comment (jbouwh) [reply source/_integrations/event.mqtt.markdown]: nice catch

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request merged**